src/firmware/app/control/trajectory_planner.h

```diff
@@ -72,232 +71,369 @@ typedef enum TrajectoryPlannerGenerationStatus
+ *
+ * @param poly  [in] The 1d polynomial representing the path along the trajectory
+ *
+ * @param segment_lengths [out] The length of segments between each node specified by the
+ * constant parameterization
+ *
+ * @param node_values [out] The evaluation of the polynomial at each value of the constant
+ * parameterization
+ *
+ * @param num_elements [in] The number of elements (nodes) to be in the trajectory
+ *
+ * Node1  |--segment length---| Node2
+ *   ->   *-------------------*  <-
+ */
+void app_trajectory_planner_generateSegmentNodesAndLengths(
+    float t_start, float t_end, Polynomial1dOrder3_t poly,
+    float segment_lengths[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float node_values[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS], unsigned int num_elements);
+
+/**
+ * Function generates the length of segments in a 2d polynomial. It works as an wrapper
+ * for the euclidean distance and values of 2d polynomials.
+ *
+ * @pre All arrays must be pre-allocated up to at least
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ *
+ * @param t_start [in] The starting value of polynomial parameterization
+ *
+ * @param t_end [in] The final value of polynomial parameterization
+ *
+ * @param poly [in] The polynomial representing the trajectory in space
+ *
+ * @param segment_lengths [out] The length of each trajectory segment (euclidean of the XY
+ * 2d polynomial)
+ *
+ * @param x_values [out] The x value of each trajectory node
+ *
+ * @param y_values [out] The y value of each trajectory node
+ *
+ * @param num_elements [in] The number of nodes(elements) of the generated trajectory
+ */
+void app_trajectory_planner_generateLinearSegmentNodesAndLengths(
+    float t_start, float t_end, Polynomial2dOrder3_t poly,
+    float segment_lengths[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float x_values[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float y_values[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS], unsigned int num_elements);
+
+/**
+ * Function generates a constant parameterization position trajectory (each node is
+ * defined by X,Y,Theta coordinates. Returns a planned trajectory with the list of
+ * guarantees based on the assumptions below
  *
  *  Key assumptions & guarantees of this planner are:
  *  - Trajectories are time-optimal assuming INFINITE JERK capability of the robot
  *
- *  - No speed on the trajectory is larger than the 'max_allowable_speed' parameter
+ *  - No speed on the trajectory is larger than the 'max_allowable_linear_speed' parameter
  *
  *  - No acceleration value between points on the trajectory can be larger than the
- * 'max_allowable_acceleration' input parameter
+ * 'max_allowable_linear_acceleration' input parameter
  *
  *  - Assuming the grip-limit of the robot IS THE SAME AS THE MAX ACCELERATION then at no
- * point on the path can the sum of centripetal and acceleration force be greater than
- * 'max_allowable_acceleration'
+ *    point on the path can the sum of centripetal and acceleration force be greater than
```

Comment:
Does the angular acceleration / orientation need to be included in this comment?

Reply:
Nope, we are assuming they are independent. The complexity introduced is quite large if we combine them.